src/rules/core/value-in-options-rule.js

```diff
@@ -65,16 +76,29 @@ module.exports = class ValueInOptionsRule extends Rule {
           for (const value of fieldValue) {
             if (allowedOptions.indexOf(value) < 0) {
               isInOptions = false;
+              isArrayButCannotBeArray = false;
               singleFieldValue = value;
               break;
             }
           }
+        } else if (fieldValue instanceof Array && !fieldObj.canBeArray()) {
+          isArrayButCannotBeArray = true;
         } else if (allowedOptions.indexOf(fieldValue) < 0) {
           isInOptions = false;
         }
       }
 
-      if (!isInOptions) {
+      if (isArrayButCannotBeArray) {
+        errors.push(
+          this.createError(
+            'fieldArray',
+            {
+              value: singleFieldValue,
+              allowedValues: `<ul><li>\`"${allowedOptions.join('"`</li><li>`"')}"\`</li></ul>`,
+            },
+          ),
```

Comment:
Note this isn't quite right, and results in this:

![Screenshot 2021-04-01 at 12 24 45](https://user-images.githubusercontent.com/2616208/113287275-45bb4900-92e5-11eb-97de-31670cb3d4c4.png)

See fix here: https://github.com/openactive/data-model-validator/commit/93af1aceaff504d0e933eecb32ae6604b5f0cbd4

Reply:
Thanks, Nick, good to know!